unix/Makefile

```diff
@@ -153,3 +160,16 @@ coverage_test: coverage
 	cd ../tests && MICROPY_MICROPYTHON=../$(DIRNAME)/micropython_coverage ./run-tests --emit native
 	gcov -o build-coverage/py ../py/*.c
 	gcov -o build-coverage/extmod ../extmod/*.c
+
+_frozen_upip.c: frozen_upip/upip.py
+	../tools/make-frozen.py frozen_upip > $@
+
+# Select latest upip version available
+UPIP_TARBALL := $(shell ls -1 -v ../tools/micropython-upip-*.tar.gz | tail -n1)
+
+frozen_upip/upip.py: $(UPIP_TARBALL)
+	rm -rf micropython-upip-*
+	tar xfz $^
+	rm -rf frozen_upip
+	mkdir -p frozen_upip
+	cp micropython-upip-*/upip*.py frozen_upip/
```

Comment:
Perhaps prefix these commands with $(Q) and print a nice message at the start like $(ECHO) "UNPACK upip".